src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -1101,44 +1187,18 @@ InstanceKlass* SystemDictionary::load_shared_class(InstanceKlass* ik,
   if (ik->has_inline_type_fields()) {
     for (AllFieldStream fs(ik); !fs.done(); fs.next()) {
       if (fs.access_flags().is_static()) continue;
+
       Symbol* sig = fs.signature();
+      int field_index = fs.index();
+
       if (fs.is_null_free_inline_type()) {
-        // Pre-load inline class
-        TempNewSymbol name = Signature::strip_envelope(sig);
-        log_info(class, preload)("Preloading class %s during loading of shared class %s. Cause: field type in LoadableDescriptors attribute", name->as_C_string(), ik->name()->as_C_string());
-        Klass* real_k = SystemDictionary::resolve_with_circularity_detection_or_fail(ik->name(), name,
-          class_loader, false, CHECK_NULL);
-        Klass* k = ik->get_inline_type_field_klass_or_null(fs.index());
-        if (real_k != k) {
-          // oops, the app has substituted a different version of k!
-          log_warning(class, preload)("Preloading of class %s during loading of shared class %s (cause: field type in LoadableDescriptors attribute) failed : %s",
-                                      name->as_C_string(), ik->name()->as_C_string(), PENDING_EXCEPTION->klass()->name()->as_C_string());
+        bool check = preload_from_null_free_field(ik, class_loader, sig, field_index, CHECK_NULL);
```

Comment:
This is a bit confusing that it both has a check for return and CHECK_NULL.  Add a comment that some preloading might fail but not fatally for the shared archive if the class changed in the app.